game/song.cc

```diff
@@ -70,18 +70,19 @@ namespace {
 
 Song::Status Song::status(double time) {
 	Note target; target.end = time;
-	Notes::const_iterator it;	
+	Notes s1, s2, notes;
+	Notes::const_iterator it;
 	if (hasDuet()) {
-		Notes s1 = getVocalTrack(0).notes;
-		Notes s2 = getVocalTrack(1).notes;
-		Notes notes;
+		s1 = getVocalTrack(0).notes;
+		s2 = getVocalTrack(1).notes;
 		std::merge(s1.begin(), s1.end(), s2.begin(), s2.end(), std::back_inserter(notes), Note::ltBegin);
 		it = std::lower_bound(notes.begin(), notes.end(), target, noteEndLessThan);
 	}
 	else {
+		notes = getVocalTrack().notes;
 		it = std::lower_bound(getVocalTrack().notes.begin(), getVocalTrack().notes.end(),target, noteEndLessThan);
```

Comment:
line 83 can be shortned with `it = std::lower_bound(notes.begin(), notes.end(),target, noteEndLessThan);`

Reply:
That's what I meant to write but it was late and I was tired and kinda frustrated. :P

Reply:
Does it already fixes this issue? if so i'll check it later tonight

Reply:
Kinda but it's far from being a complete solution, ergo why it's still [WIP]